docs/debugging-dotnet-lambda-functions.md

```diff
@@ -0,0 +1,149 @@
+                        "pipeProgram": "powershell",
+                        "pipeArgs": [
+                            "-c",
+                            "docker exec -i $(docker ps -q -f publish=5679) ${debuggerCommand}"
+                        ],
+                        "debuggerPath": "/tmp/lambci_debug_files/vsdbg",
+                        "pipeCwd": "${workspaceFolder}/path/to/project",
+                    }
+                },
+
+                "sourceFileMap": {
+                    "/var/task": "${workspaceFolder}/path/to/project"
+                }
+            }
+        ]
+    }
+    ```
+
+2. Launch Visual Studio Code and open the folder containing your application.
+3. Press `Ctrl+Shift+D` or click the `Debug` icon to open the debug viewlet:
+
+    ![Debug Icon](./images/view_debug.png)
+
+4. Select `.NET Core: Remote Attach` from the drop-down menu at the top of the viewlet.
+
+## Start debugging
+
+1. Set a breakpoint anywhere in your lambda handler.
+2. Open a terminal in the folder containing `template.yaml`, and run the following commands. The SAM CLI will invoke your lambda handler, and wait for a debugger to attach to it.
+
+    * Replace `HelloWorldFunction` with the name of the function that you want to invoke.
+    * Replace `5679` with the port that you specified in `launch.json`.
+
+    ```bash
+    sam build # TODO: First set the environment variable that enables building the Debug configuration.
+    sam local invoke HelloWorldFunction -d 5679 --debugger-path path/to/project/.vsdbg --no-event
+    ```
+
+3. When you see `Waiting for debugger to attach...`, go back to Visual Studio Code and press F5 to attach the debugger to the handler that you invoked in the previous step.
+
+## Optional: Automatically start debugging when ready
+
+With the above steps, you need to manually invoke SAM CLI from the command line, wait for it to be ready, then attach the debugger. We can automate the process of invoking SAM CLI and waiting for it to be ready by using a `preLaunchTask`.
+
+1. Open `<sam app root>/.vscode/tasks.json` (create a new file if it does not already exist).
+2. Add the following contents to `tasks.json`:
+
+    * Replace `HelloWorldFunction` with the function that you wish to debug.
+    * Replace `path/to/project` with the relative path to the directory containing the `.csproj` project that you wish to debug.
+
+    ```jsonc
+    {
+        "version": "2.0.0",
+        "tasks": [
+            {
+                "label": "Debug .NET Core Lambda Function",
+                "type": "shell",
+                "command": "sam",
+                "args": [
+                    "local",
+                    "invoke",
+                    "HelloWorldFunction",
+                    "-d",
+                    "5679",
+                    "--debugger-path",
+                    "path/to/project/.vsdbg",
+                    "--no-event"
+                ],
+                "isBackground": true,
```

Comment:
I also had to add the following for this to work:
```jsonc
            "options": {
                "cwd": "${workspaceFolder}/path/to/folder-containing-samtemplate"
            },
```

Reply:
Updated the example configuration to avoid depending on ${workspaceFolder}. Let me know if you still see this issue.

Reply:
In its current form the sam cli is unable to locate the template.yaml file. The options.cwd section needs to be set to the location of the template.yaml file.

Reply:
I accidentally resolved this conversation. This still needs to be changed (see previous comment)